src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java

```diff
@@ -193,5 +237,32 @@ protected static QueryOuterClass.Query bindUpdateValues(
     return QueryOuterClass.Query.newBuilder(builtQuery).setValues(values).build();
   }
 
-  record UpdatedDocument(DocumentId id, boolean upserted, JsonNode document) {}
+  private Uni<ReadDocument> readDocumentAgain(
+      QueryExecutor queryExecutor, ReadDocument prevReadDoc) {
+    // Read again if retry flag is `true`
+    return readOperation()
+        .getDocuments(
+            queryExecutor,
+            null,
+            new DBFilterBase.IDFilter(DBFilterBase.IDFilter.Operator.EQ, prevReadDoc.id()))
+        .onItem()
+        .transform(
+            response -> {
+              if (!response.docs().isEmpty()) {
+                return response.docs().get(0);
+              } else {
+                // If data changed and doesn't satisfy filter conditions
+                return null;
+              }
+            });
+  }
+
+  record UpdatedDocument(DocumentId id, boolean upserted, JsonNode document, Throwable error) {}
+
+  /** Inherited Exception class to handle retry */
+  public class LWTException extends JsonApiException {
```

Comment:
Is there a duplicate `LWTException` in `DeleteOperation`, to be deleted? Might make sense to extract this exception class since it is public.

Reply:
Moved to outer class with 2 different constructor, will make it to one when I work on delete error message handling,